libnymea-core/hardware/network/networkdevicediscoveryimpl.cpp

```diff
@@ -247,6 +247,9 @@ void NetworkDeviceDiscoveryImpl::unregisterMonitor(const MacAddress &macAddress)
 
 void NetworkDeviceDiscoveryImpl::unregisterMonitor(NetworkDeviceMonitor *networkDeviceMonitor)
 {
+    if (!networkDeviceMonitor)
```

Comment:
This probably doesn't do what you intended... This would only catch the case where the user passes in a nullptr, but it would still crash if the user calls this twice on the same monitor.

Reply:
Until we are not giving a dedicated object to each reequester, this should at least fix situations like: `unregisterMonitor(m_monitors.take(thing))` within plugins where the monitor is already removed from the hash or stuff...